### DISCUSSION TITLE
Topology view not showing

### DISCUSSION THREAD

**Discussion by NULLProtogen** [mattieserver/netbox-topology-views#208] [Q&A]
Possible user error but I can't get the topology views plugin to show any views.

I'm not sure what logs or configs I can provide so please let me know what you would like.

PLUGINS = ["netbox_topology_views"]
PLUGINS_CONFIG = {
    'netbox_topology_views': {
        'enable_circuit_terminations': False,
        'draw_default_layout': True,
        'ignore_cable_type': 'circuit termination'
    }
}

Python 3.8.10
Netbox 3.2.5

Installed Plugins
NAME     PACKAGE NAME     AUTHOR     DESCRIPTION     VERSION
Topology views	netbox_topology_views	Mattijs Vanhaverbeke	An plugin to render topology maps	2.0.1

**Comment by jaydabi** [comment]
Same issue here. Setting `Hide Unconnected` filter makes the nodes appear.

**Comment by mattieserver** [comment]
Hi,

Sorry for the late reply.
Can you check if you have this issue still on netbox 3.27?
I was unable to replicatie this, but it is not so easy to get the logs from netbox.

**Comment by hardaker** [comment]
I'm on 3.27 and haven't gotten anything to display yet.

**Comment by hardaker** [comment]
So it turns out that the problem is that the static js/css/etc files are not getting installed in my docker version of netbox.  Mounting the source code's static code into `/opt/netbox/netbox/static/netbox_topology_views` in the docker container let it work.

**Comment by mattieserver** [comment]
Hi @hardaker,

Do you have the `python3 manage.py collectstatic --no-input` command in your docker build process?
If you run this after you install the plugin it should copy the css  & js files to the correct location.

**Comment by hardaker** [comment]
Sorry for the delay.  I do.  But your point is valid: why isn't it collecting it?  I'll try to look into it as well.

**Comment by dwilliss** [comment]
@hardaker were you able to find a solution?  I'm having (was having) the same problem.
I also see that there's no `network-topology-views` in the static folder.
I added the following to my Dockerfile

`COPY ./netbox_topology_views/static/netbox_topology_views /opt/netbox/netbox/static/netbox_topology_views`

And now I have icons

**Comment by hardaker** [comment]
I admittedly have been swamped with a much more pressing deadline and haven't gotten back to debugging.  But yes, copying in solved it for me too and hence motivation is low :-/

**Comment by andrewm659** [comment]
This is not working for me on netbox-3.3.1.

I'm not seeing anything in the logs...

**Comment by mattieserver** [comment]
Did you clear your browser cache after the update?

**Comment by andrewm659** [comment]
Stupid me.  I did not but just tried it using Vivaldi.  No dice after clearing cache.  Will try another browser and computer.

**Comment by andrewm659** [comment]
Just tried on a different computer.  Nothing comes up.  Even when I check Hide Unconnected

**Comment by mattieserver** [comment]
Is this also with a docker setup?

**Comment by andrewm659** [comment]
No.  Not using docker.

**Comment by rizlas** [comment]
This is currently not working in a docker environment as already pointed out from @hardaker and @dwilliss .

```
RUN /opt/netbox/venv/bin/pip install --no-warn-script-location -r /tmp/externals_plugins.txt && \
    # Copy static or plugin will not work
    cp -r /opt/netbox/venv/lib/python3.10/site-packages/netbox_topology_views/static/netbox_topology_views /opt/netbox/netbox/static && \
    /opt/netbox/netbox/manage.py collectstatic --no-input
```

Result in 

> 🧬 loaded config '/etc/netbox/config/configuration.py'
> 🧬 loaded config '/etc/netbox/config/extra.py'
> 🧬 loaded config '/etc/netbox/config/logging.py'
> 🧬 loaded config '/etc/netbox/config/plugins.py'
> 
> 0 static files copied to '/opt/netbox/netbox/static', 514 unmodified.

**Comment by leijonatm** [comment]
Hello all, 
I tried to use the topology views plugin today in the latest version of netbox and still experience the same issue i.e. after running the `docker compose build--no-cache` command

> % docker compose build --no-cache
[+] Building 0.0s (2/2) FINISHED                                                                 docker:desktop-linux
 => [netbox internal] load .dockerignore                                                                         0.0s
 => => transferring context: 162B                                                                                0.0s
 => [netbox internal] load build definition from Dockerfile-Plugins                                              0.0s
 => => transferring dockerfile: 624B                                                                             0.0s
failed to solve: dockerfile parse error on line 10: unknown instruction: /opt/netbox/netbox/manage.py

and here are the contents of the docker file

> % cat Documents/docker/netbox-docker/Dockerfile-Plugins
FROM netboxcommunity/netbox:latest
COPY ./plugin_requirements.txt /opt/netbox/
RUN /opt/netbox/venv/bin/pip install  --no-warn-script-location -r /opt/netbox/plugin_requirements.txt
COPY configuration/configuration.py /etc/netbox/config/configuration.py
COPY configuration/plugins.py /etc/netbox/config/plugins.py
COPY ./netbox_topology_views/static/netbox_topology_views /opt/netbox/netbox/static/netbox_topology_views
RUN SECRET_KEY="dummydummydummydummydummydummydummydummydummydummy" /opt/netbox/venv/bin/python
/opt/netbox/netbox/manage.py collectstatic --no-input

Any idea why I am unable to use the plugin ?

**Reply by mattieserver** [reply]
Please just open a new issue/discussion. Its not even the same error.

**Reply by leijonatm** [reply]
What I am experiencing is what the title saying, topology view not showing. What else shall I describe in the new issue discussion ?

**Reply by leijonatm** [reply]
My bad ... I just remembered I commented out the last line in the DockerFile which installed the plugin but it is not showing. Forget my last comment. 
Sorry ...

**Comment by leijonatm** [comment]
@mattieserver I was able to build the container following the instructions for Docker install although I now am facing the same issue as the title. 

What am I missing now ?

**Comment by rlljorge** [comment]
Hi ! 

I have them similar problem, the devices is not show up, setting Hide Unconnected not works for me.

Python 3.10.12
Netbox v3.6.4
netbox-topology-views 3.8.0
Ubuntu 22.04.3 LTS without Docker 

Best Regards,

Rodrigo

**Reply by joaolucasmacedo** [reply]
I have the same issue. 
Fresh install of the plugin, Netbox 3.6.4 ans Python 3.10
No Docker.

**Reply by mattieserver** [reply]
Please show what options you have selected on the topology page.

**Reply by joaolucasmacedo** [reply]
My preferences:
![image](https://github.com/mattieserver/netbox-topology-views/assets/66760118/33e6bdcc-3612-4440-9a2d-1e3c3241c078)

Default page:
![image](https://github.com/mattieserver/netbox-topology-views/assets/66760118/4943675e-29aa-4fe0-9600-be178443cd00)

Also, looks like it does not apply the filters I select.  Example:
I've selected "show cables", "show unconnected" and a site. After click search only the site shows on the filter.

**Reply by joaolucasmacedo** [reply]
As the option "Draw default layout is selected, it show all devices when I click the topology menu, but if I use anything to filter nothing happens.

**Reply by AzRe3aL** [reply]
I am seeing the same, selections in filters don't seem to have an effect after hitting "search"
Only  "Draw default layout" seems to display anything.

Fresh install of the plugin, Netbox 3.6.4 Docker

default page 
![image](https://github.com/mattieserver/netbox-topology-views/assets/91475604/b07706c4-4f6e-432f-88a0-8eb984bc0b3c)

![image](https://github.com/mattieserver/netbox-topology-views/assets/91475604/04cdffb6-ca35-4d5d-9212-18b251b1ef3f)

**Reply by dreng** [reply]
Please note that this is fixed in the development branch already. See #408

**Comment by alemsas** [comment]
This is also my problem!
I installed the topology view with [this](https://github.com/netbox-community/netbox-docker/wiki/Using-Netbox-Plugins)
But in my topology view page I didnt see my devices!My cable connections are ok!and I am sure with this!
also I try this

```
python3 manage.py collectstatic --no-input
🧬 loaded config '/etc/netbox/config/configuration.py'
🧬 loaded config '/etc/netbox/config/extra.py'
🧬 loaded config '/etc/netbox/config/logging.py'
🧬 loaded config '/etc/netbox/config/plugins.py'

0 static files copied to '/opt/netbox/netbox/static', 570 unmodified.
```
```
cat configuration/plugins.py
PLUGINS = ["netbox_topology_views"]
```
I use docker for this.

**Comment by deanfourie1** [comment]
I am having the same issue, I just see a completely blank screen, Not using docker and on latest version to date.

![image](https://github.com/user-attachments/assets/4448e814-4594-43d6-ba14-ab972860f28e)

**Reply by mattieserver** [reply]
You have to specify filters. Please open the filter tab and select what you want to see.

**Reply by deanfourie1** [reply]
Ok thanks that worked. I will have a play.

**Reply by deanfourie1** [reply]
Is there a way to save a default filter to show by default? Thanks

**Reply by dreng** [reply]
> Is there a way to save a default filter to show by default? Thanks

No, not yet. This would be possible to implement. That said, I would prefer if this was a feature of NetBox itself.

**Reply by deanfourie1** [reply]
That's weird, because before I updated to 4.0, the topology view would always just show, I never needed to set the filters.

Now it seems that I NEED to set the filters in order to see the topology.

Is this because of the 4.0 update?

Thanks

**Reply by dreng** [reply]
> Is this because of the 4.0 update?

Nothing has changed regarding this feature. It is possible to draw the topology on the initial load. Make sure you have checked the appropriate option in "Individual Options".
